images_config/ks-robot-master.cfg

```diff
@@ -106,71 +106,72 @@ update-grub
-%post --interpreter=/bin/bash
-!/bin/bash
-sh -c 'echo "deb http://packages.ros.org/ros/ubuntu trusty main" >> /etc/apt/sources.list.d/ros-latest.list'
-wget http://packages.ros.org/ros.key -O - | apt-key add - 
-apt-get update
-apt-get install ros-indigo-roscpp -y
-apt-get install ros-indigo-rosbash -y
-apt-get install ros-indigo-ros -y
-apt-get install ros-indigo-care-o-bot-robot -y
+################ ADD USERS ################ 
+%post --interpreter /bin/bash
+echo "::Adding users and groups ::"
+echo "session	required	pam_mkhomedir.so " >> /etc/pam.d/common-session
+################ ROBOT-LOCAL USER ################ 
+sed -i 's/HOME=/home/HOME=/iscsi/user/g' /etc/default/useradd
+adduser --disabled-password --gecos "" robot-local --home /home/robot-local
+echo "robot-local:sawVsPPn2.KXM"  | chpasswd -e
+cp -rT /etc/skel /home/robot-local #to populate /home/robot-local with default files and folders
+chown robot-local:robot-local /home/robot-local
+################ ROBOT USER ################ 
+#sudo useradd -m -p $1$g1R/.PH0$d6gtcznTWbNqtRq3SSerH1 -d /u/robot robot
+adduser --disabled-password --gecos "" robot --home /u/robot
+echo "robot:sawVsPPn2.KXM"  | chpasswd -e
+usermod -m -d /u/robot robot
+cp -rT /etc/skel /u/robot/#to populate /u/robot with default files and folders
+chown robot:robot /u/robot/ #to change the owner of /u/robot to user robot
 %end
 
-############################NETDATA TOOLS##############################################
-%post --interpreter /bin/bash
-echo -e "\n${green}INFO: NETDATA TOOLS${NC}\n"
-apt-get install zlib1g-dev uuid-dev libmnl-dev gcc make git autoconf autoconf-archive autogen automake pkg-config curl -y
-git clone https://github.com/firehol/netdata.git --depth=1 /root/netdata
-cd /root/netdata
-yes | ./netdata-installer.sh
+################ SETUP ROOT USER ################ 
+%post
+echo "Defaults rootpw" >> /etc/sudoers
+passwd root --iscrypted sawVsPPn2.KXM
+%end
+
+################ GIVE ROBOT-LOCAL AND ROBOT USERS FULL RIGHTS ################ 
+%post
+echo "robot-local ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
+echo "robot ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
 %end
-#######################################################################################
-############## NFS SETUP #############
+
+################ KEYBOARD LAYOUT ################ 
+%post --interpreter=/bin/bash 
+L='de' && sed -i 's/XKBLAYOUT=\"\w*"/XKBLAYOUT=\"'$L'\"/g' /etc/default/keyboard
+apt-get install console-data -y -f
+dpkg-reconfigure keyboard-configuration
+%end
+
+################ NFS SETUP ################ 
 %post --interpreter /bin/bash
-echo -e "\n${green}INFO:  Install NFS${NC}\n"
 sleep 5
 apt-get install nfs-kernel-server autofs -y -f
 echo "/home /u none bind 0 0" >> /etc/fstab
 echo "/u *(rw,fsid=0,sync,no_subtree_check)" >> /etc/exports
 { rm /etc/default/nfs-common && awk '/^NEED_STATD=/ {sub(/\=/,"=yes")}1' > /etc/default/nfs-common; } < /etc/default/nfs-common 
```

Comment:
I have tried many ways to make NEED_STATD=yes which was on NEED_STATD in the file of /etc/default/nfs-common , its more customized format which removes and replace same file with required changes